### PR TITLE
Add new category icons

### LINKED ISSUE #2778: Implement new category icons
~~We need editorial copy for all categories so we can display things that look like:~~

This is now just to update to the new Icons. The icons are available in https://github.com/mozilla/addons-frontend/issues/2621

screenshot 2017-07-12 02 03 16

----

Categories we need to replace the icons for:

#### Extensions

* [x] Language Support
* [x] Feeds, News & Blogging
* [x] Download Management
* [x] Alerts & Updates
* [x] Bookmarks
* [x] Appearance
* [x] Games & Entertainment
* [x] Other
* [x] Photos, Music & Videos
* [x] Privacy & Security
* [x] Search Tools
* [x] Shopping
* [x] Social & Communication
* [x] Tabs
* [x] Web Development

#### Themes

* [x] Abstract
* [x] Causes
* [x] Fashion
* [x] Film and TV
* [x] Firefox
* [x] Foxkeh
* [x] Holiday
* [x] Music
* [x] Nature
* [x] Other
* [x] Scenery
* [x] Seasonal
* [x] Solid
* [x] Sports
* [x] Web Sites

### PR DESCRIPTION
**Pull request by willdurand**
Fix #2778 

---

This PR updates the new category header with icons. I kept the existing description as it looks nicer with it rather than without it (as someone mentioned in the issue). It is hidden on small screens though.

In order to color the SVG icons, I use `mask`, which is rather well supported now: http://caniuse.com/#feat=css-masks.

## Screenshots

Extensions:

screen shot 2017-09-28 at 17 20 52

Themes:

screen shot 2017-09-28 at 17 20 46

Small screens:

screen shot 2017-09-28 at 17 21 14

screen shot 2017-09-28 at 17 30 10

### CODE REVIEW

**Comment by tofumatt** [comment]
Woohoo, masks!

**Comment by willdurand** [comment]
I tested on Edge, I misread the caniuse page actually. It is still "under consideration" on Edge, while all other browsers support it! I have fixed the issue for Edge by rendering a degraded version of the icon (i.e. without color).

See:

bs_win10_edge_15 0 1
bs_win10_edge_15 0

File: src/amo/components/CategoryIcon/styles.scss [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+$default-icon-size: 64px;
+
+@mixin icon($name) {
+  display: inline-block;
+  height: $default-icon-size;
+  mask: url("./#{$name}.svg") no-repeat 50% 50%;
+  transform: scale(0.8);
+  width: $default-icon-size;
+
+  // Edge does not support CSS mask, but all other browsers do. This avoids
```
Comment (tofumatt) [inline src/amo/components/CategoryIcon/styles.scss]: I would rephrase this comment: "This prevents Edge from rendering a square block of color because it can't mask over the SVG."
